ygen/gogen.go

```diff
@@ -1010,6 +1056,60 @@ func (t *{{ .ParentReceiver }}) To_{{ .Name }}(i interface{}) ({{ .Name }}, erro
 		]", i, i)
 	}
 }
+`)
+
+	// unionTypeSimpleTemplate outputs the type that corresponds to a multi-type union
+	// in the YANG schema. It does so by enhancing generated typedefs.
+	unionTypeSimpleTemplate = mustMakeTemplate("unionTypeSimple", `
+// {{ .Name }} is an interface that is implemented by valid types for the union
+// for the leaf {{ .LeafPath }} within the YANG schema.
+// Union type can be one of [{{ .SubtypeDocumentation }}].
+type {{ .Name }} interface {
+	// Union type can be one of [{{ .SubtypeDocumentation }}]
+	Documentation_for_{{ .Name }}()
```

Comment:
`godoc` doesn't seem to treat this comment any differently -- on the webpage the comment appears as-is, even when the name doesn't start with the name of the function.

The purpose of this different name is to assist with users who might be using autocomplete to figure out a particular union's subtypes. e.g. if the user doesn't know the union subtypes of A_B_Union, they can type out in their IDE A_B_Union, click on this documentation string, and see what the subtypes are.

Reply:
Ah, I understand. I was wondering whether there was something specific in terms of calling this `Documentation...` maybe just make it `{{ .Name }}_Doc` since the most important thing seems to be the existence of the method rather than the name?

Reply:
For a user who is confused, the `Documentation_...` is an obvious thing to read. Putting `_Doc` last might get hidden behind a really long union name, which contains the entire path of the node. Also, `Doc_{{ .Name }}` is not as clear as `Documentation_for_{{ .Name }}` as to the purpose of the method.

Reply:
Sure, that SGTM.